x/concentrated-liquidity/swaps_test.go

```diff
@@ -2091,6 +2091,9 @@ func (s *KeeperTestSuite) TestComputeAndSwapOutAmtGivenIn() {
 				s.assertPoolNotModified(poolBeforeCalc)
 			}
 
+			pool, err = s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, pool.GetId())
+			s.Require().NoError(err)
```

Comment:
Same question with refetching these pool objects again, if we need to refetch multiple times why cant we just operate on the pointer?

Reply:
we refetch because the pool obj change after fist position creation. And all most of these calls are in different test functions, with their own independent logic so i'm not sure how helpful pointer will be